frontend/expr_inferrer.py

```diff
@@ -434,21 +321,45 @@ def infer_dict_comprehension(node, context):
     infer_generators(node.generators, local_context)
     key_type = infer(node.key, local_context)
     val_type = infer(node.value, local_context)
-    return TDictionary(key_type, val_type)
+    return z3_types.Dict(key_type, val_type)
+
+
+def _get_args_types(args, context):
+    """Return inferred types for function call arguments"""
+    # TODO kwargs
+    if len(args) > 5:
+        raise NotImplementedError("Functions with more than 5 arguments are not yet supported.")
```

Comment:
Could you store the largest number of arguments you encountered, and then generate the necessary stuff up until that number?